### PR TITLE
Add Content Security Policy

### PR DESCRIPTION
**Pull request by nicoonoclaste**

### CODE REVIEW

**Comment by nicoonoclaste** [comment]
`search/main.js` also needs fixing (after the inline data change), but since we don't really need it right now (and I don't feel like making a “custom” search plugin just to fix it), I'll just fix it when I send the patches upstream.
Update: fixed in dbc8be8

**Comment by nicoonoclaste** [comment]
Upstreaming in https://github.com/mkdocs/mkdocs/pull/1907

**Comment by nicoonoclaste** [comment]
> How come `style-src` is pointing at a js file?

That URI (`https://cdnjs.cloudflare.com/ajax/libs/highlight.js/`) covers both Javascript and CSS files:
- `https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js`
- `https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/github.min.css`

**Review by reuvenpo — APPROVED** [review]

**Comment by nicoonoclaste** [comment]
This will actually need updating, if we load more things from CDNJS.

I'll update in a moment and request a re-review  :3

**Comment by nicoonoclaste** [comment]
@reuvenpo Could you re-review?  Here is what changed:
- I redid the changes to use the TOML config syntax instead, so the CSP should be more readable.
- Since we excised the binaries from the repo (and load them from CDNJS instead), I had to update the CSP to whitelist more libraries from CDNJS.

File: netlify.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[[headers]]
+	for = "/*"
+	[headers.values]
+		# Require use of HTTPS on the website
+		Strict-Transport-Security = "max-age=63072000"
```
Comment (reuvenpo) [inline netlify.toml:6]: I'd add a comment describing the arithmetic here. How was this figure calculated?

File: netlify.toml [hunk 2]
```diff
@@ -4,6 +4,25 @@
...
+		#   - same origin (for images, scripts, and styles) ;
+		#   - CDNJS, for a certain whitelist of libraries.
+		# - Forbid loading the site in an iframe (can be abused for clickjacking)
+		Content-Security-Policy = """
+		default-src 'none'; img-src 'self';
+		font-src   https://cdnjs.cloudflare.com/ajax/libs/font-awesome/;
+		style-src 'self'
+		           https://cdnjs.cloudflare.com/ajax/libs/font-awesome/
+		           https://cdnjs.cloudflare.com/ajax/libs/highlight.js/
+		           https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/;
```
Comment (reuvenpo) [inline netlify.toml]: I'd use less indentation and put `'self'` on its own line, but that's just style so feel free to dismiss this comment

File: netlify.toml [hunk 3]
```diff
@@ -27,6 +28,8 @@
 		base-uri        'none';
 		form-action     'none';
 		frame-ancestors 'none';
+		sandbox allow-forms allow-modals allow-popups
+		        allow-same-origin allow-scripts;
```
Comment (reuvenpo) [inline netlify.toml]: Similarly to the comment above, I'd put each `allow-x` here on it's own line.

**Review by reuvenpo — APPROVED** [review]

**Pull request merged**